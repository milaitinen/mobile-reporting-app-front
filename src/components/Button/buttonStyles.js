import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
    view: {
        backgroundColor: 'transparent',
        borderRadius: '$buttonBorderRadius',
        justifyContent: 'center',
        alignContent: 'center',
        height: 70,
        marginHorizontal: 10,
        marginBottom: 2,
    },

    sendButtonContainer: {
        backgroundColor: '#99d9ad',
        borderRadius: '$buttonBorderRadius',
        borderWidth: 1,
        alignContent: 'center',
        justifyContent: 'center',
        paddingLeft: 15,
        paddingRight: 15,
        borderColor: '#99d9ad',
        '@media (max-width: 350)': {
            borderRadius: '$buttonBorderRadiusSmall',
            height: 30,
        },
        '@media (min-width: 350)': {
            height: 40,
        },
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        },
    },

    saveButtonContainer: {
        backgroundColor: '$draftBlue',
        borderRadius: '$buttonBorderRadius',
        borderWidth: 1,
        alignContent: 'center',
        justifyContent: 'center',
        paddingLeft: 15,
        paddingRight: 15,
        borderColor: '$draftBlue',
        '@media (max-width: 350)': {
            borderRadius: '$buttonBorderRadiusSmall',
            height: 30,
        },
        '@media (min-width: 350)': {
            height: 40,
        },
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        },
    },

    buttonContent: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignContent: 'center',
    },

    text: {
        paddingLeft: 10,
        color: '#474c52',
        fontFamily: '$primaryFont',
        textAlign: 'center',
        '@media (max-width: 350)': {
            fontSize: 12,
        },
        '@media (min-width: 350)': {
            fontSize: 16,
        },
    },
});

export default styles;