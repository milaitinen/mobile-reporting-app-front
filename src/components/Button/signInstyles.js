import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
    view: {
        flex: 0.7,
        flexDirection: 'row',
        justifyContent: 'center',
    },

    buttonContainer: {
        backgroundColor: '#9dcbe5',
        borderRadius: '$buttonBorderRadius',
        borderWidth: 1,
        alignContent: 'center',
        justifyContent: 'center',
        paddingLeft: 15,
        paddingRight: 15,
        borderColor: '#9dcbe5',
        '@media (max-width: 350)': {
            borderRadius: '$buttonBorderRadiusSmall',
            marginTop: 20,
            height: 30,
        },
        '@media (min-width: 350)': {
            marginTop: 30,
            marginBottom: 60,
            height: 40,
        }
    },
    text: {
        color: '#274752',
        fontFamily: '$primaryFont',
        textAlign: 'center',
        '@media (max-width: 350)': {
            fontSize: 12,
        },
        '@media (min-width: 350)': {
            fontSize: 16,
        }
    },
});

export default styles;