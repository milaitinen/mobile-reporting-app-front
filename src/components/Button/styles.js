import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({

    buttonContainer: {
        backgroundColor: '#9dcbe5',
        marginTop:30,
        paddingTop:10,
        paddingBottom:10,
        marginLeft:115,
        marginRight:115,
        marginBottom: 60,
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#9dcbe5',
    },
    text: {
        color: '#274752',
        fontFamily: '$primaryFont',
        fontSize: 15,
        textAlign: 'center',
    },
});

export default styles;