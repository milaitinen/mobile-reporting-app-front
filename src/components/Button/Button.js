import React from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import color from 'color';
import { Icon } from 'react-native-elements';
import styles from './buttonStyles';

// TODO: styling
const CustomButton = ({ title, type, onPress }) => {

    if (type === 'send') {
        return (
            <View style={styles.view}>
                <TouchableHighlight
                    underlayColor={color('#9dcbe5').darken(0.1)}
                    style={styles.sendButtonContainer}
                    onPress={onPress}>
                    <View style={styles.buttonContent}>
                        <Icon name={'send'} color={'#474c52'} size={18}/>
                        <Text style={styles.text}>{title}</Text>
                    </View>
                </TouchableHighlight>
            </View>
        );
    }
    return (
        <View style={styles.view}>
            <TouchableHighlight
                underlayColor={color('#8cc9e5').darken(0.1)}
                style={styles.saveButtonContainer}
                onPress={onPress}>
                <View style={styles.buttonContent}>
                    <Icon name={'save'} color={'#474c52'} size={18}/>
                    <Text style={styles.text}>{title}</Text>
                </View>
            </TouchableHighlight>
        </View>
    );
};

export default CustomButton;