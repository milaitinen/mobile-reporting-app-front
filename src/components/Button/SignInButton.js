import React from 'react';
import { Text, TouchableHighlight } from 'react-native';

import styles from './styles';

const SignInButton = ({ onPress, children }) => (
    <TouchableHighlight
        onPress={onPress}
        style={styles.touchable}
    >
        <Text style={styles.text}>{children}</Text>
    </TouchableHighlight>
);

export default SignInButton;