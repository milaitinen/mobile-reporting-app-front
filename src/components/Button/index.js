import SignInButton from './SignInButton';
import Button from './Button';
import styles from './signInstyles';

export { Button, SignInButton, styles };