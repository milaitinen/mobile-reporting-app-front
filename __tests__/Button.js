import { TouchableHighlight } from 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Button from '../src/components/Button/Button';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer.create(
        <Button title='some button' type={'send'} onPress={jest.fn()} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders only one text field when dateAccepted = null', () => {
    expect(shallow(<Button title='some button' type={'save'} onPress={jest.fn()} />).find(TouchableHighlight).length).toEqual(1)
});




