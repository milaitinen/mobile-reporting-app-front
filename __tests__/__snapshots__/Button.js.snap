// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={undefined}
    >
      <View
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#474c52",
                "fontSize": 18,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          î…£
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        some button
      </Text>
    </View>
  </View>
</View>
`;
